### PR TITLE
Feature/geo json bounding box

### PR DESCRIPTION
**Pull request by rsupnekar**
Fixed for multipolygons and features not starting with 'feature'

### CODE REVIEW

File: sdk/sdk-api/src/main/java/mil/emp3/json/geoJson/GeoJsonParser.java [hunk 1]
```diff
@@ -43,6 +43,7 @@
             JsonValue value = Json.parse(reader);
             return parseJsonObject(value);
         } catch (Exception ioe) {
+            ioe.printStackTrace();
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline sdk/sdk-api/src/main/java/mil/emp3/json/geoJson/GeoJsonParser.java:48]: > ioe.printStackTrace();

Likely is not needed. Assumption would be that whomever is invoking this method would handle either the exception or log it or something else.

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java [hunk 2]
```diff
@@ -1593,18 +1593,33 @@ public void captureFailed(Exception Ex) {
...
-                    camera.setLatitude(0);
-                    camera.setLongitude(20);
-                    camera.setAltitude(3e6);
+                    camera.setLatitude(44.5);
+                    camera.setLongitude(1);
+                    camera.setAltitude(5e5);
                     camera.apply(false);
-                } catch (EMP_Exception e) {
+                } catch (IOException|EMP_Exception e) {
                     e.printStackTrace();
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:1624]: >e.printStackTrace();

Likely better to throw the exception than trap it.

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java [hunk 3]
```diff
@@ -1593,18 +1593,33 @@ public void captureFailed(Exception Ex) {
...
+                        this.oFeatureHash.put(feature.getGeoId(), feature);
+                    }
+                    stream.close();
                     ICamera camera = this.map.getCamera();
-                    camera.setLatitude(0);
-                    camera.setLongitude(20);
-                    camera.setAltitude(3e6);
+                    camera.setLatitude(44.5);
+                    camera.setLongitude(1);
+                    camera.setAltitude(5e5);
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:1621]:
```
camera.setLatitude(44.5);
camera.setLongitude(1);
camera.setAltitude(5e5);
```

Do these coordinates represent anything in particular?

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java [hunk 4]
```diff
@@ -1593,18 +1593,33 @@ public void captureFailed(Exception Ex) {
...
                 try {
                     // Sample geoJSON from CMAPI 1.2 document, edited for correctness
-                    InputStream stream = getApplicationContext().getResources().openRawResource(R.raw.cmapi);
+                    InputStream stream = getApplicationContext().getResources().openRawResource(R.raw.communes_69);
                     List<IFeature> featureList = GeoJsonParser.parse(stream);
                     for (IFeature feature : featureList) {
                         this.oRootOverlay.addFeature(feature, true);
                         this.oFeatureHash.put(feature.getGeoId(), feature);
                     }
+                    stream.close();
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:1603]: > stream.close();

Better to have in finally block.

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java [hunk 5]
```diff
@@ -1593,18 +1593,33 @@ public void captureFailed(Exception Ex) {
...
+                    stream = getApplicationContext().getResources().openRawResource(R.raw.random_geoms);
+                    featureList = GeoJsonParser.parse(stream);
+                    for (IFeature feature : featureList) {
+                        this.oRootOverlay.addFeature(feature, true);
+                        this.oFeatureHash.put(feature.getGeoId(), feature);
+                    }
+                    stream.close();
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:1610]: > stream.close();

Better to have in finally block.

File: tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java [hunk 6]
```diff
@@ -1593,18 +1593,33 @@ public void captureFailed(Exception Ex) {
...
+                    stream = getApplicationContext().getResources().openRawResource(R.raw.rhone);
+                    featureList = GeoJsonParser.parse(stream);
+                    for (IFeature feature : featureList) {
+                        this.oRootOverlay.addFeature(feature, true);
+                        this.oFeatureHash.put(feature.getGeoId(), feature);
+                    }
+                    stream.close();
```
Comment (ee11cbb19052e40b07aac0ca060c23ee) [inline tests/basic/src/main/java/mil/emp3/dev_test_sdk/MainActivity.java:1617]: > stream.close();

Better to have in finally block.

**Review by ee11cbb19052e40b07aac0ca060c23ee — CHANGES_REQUESTED** [review]
Improve exception handling and include comments for magic numbers (the coordinate values).

**Review by Ish-rivera — COMMENTED** [review]
Fix stream closing.

**Review by ee11cbb19052e40b07aac0ca060c23ee — APPROVED** [review]

**Pull request merged**